go/kbfs/libkbfs/folder_block_ops.go

```diff
@@ -1485,6 +1487,25 @@ func (fbo *folderBlockOps) updateEntryLocked(ctx context.Context,
 	} else {
 		_ = fbo.makeDirDirtyLocked(lState, parentPath.TailPointer(), unrefs)
 	}
+
+	// If we're in the middle of syncing the directories, but the
+	// current file is not yet being synced, we need to re-apply this
+	// update after the sync is done, so it doesn't get lost after the
+	// syncing directory block is readied.  This only applies to dir
+	// updates being caused by file changes; other types of dir writes
+	// are protected by `folderBranchOps.syncLock`, which is held
+	// during `SyncAll`.
+	if fbo.dirtyDirsSyncing && !fbo.doDeferWrite {
+		fbo.log.CDebugf(ctx, "Deferring update entry during sync")
+		n := fbo.nodeCache.Get(file.TailRef())
+		fbo.deferredDirUpdates = append(
+			fbo.deferredDirUpdates, func(lState *kbfssync.LockState) error {
+				file := fbo.nodeCache.PathFromNode(n)
```

Comment:
Is there a chance this will no longer be in the cache?

Reply:
Nope -- it only gets garbage collections when there are no more references to it, and the closure itself is holding a reference at least.  Plus the node for the file being written is held by the caller somewhere, and it will have a reference to its parent as well.